packages/near-fast-auth-signer/src/lib/useAuthState.ts

```diff
@@ -22,21 +23,28 @@ export const useAuthState = (skipGetKeys = false): AuthState => {
   }, []);
 
   const [controllerState, setControllerState] = useState<'loading' | boolean>('loading');
+  const [isPassKeySupported, setIsPassKeySupported] = useState<boolean>(null);
+
+  useEffect(() => {
```

Comment:
Do we need an extra useEffect and useState? 

I think we could to this fetch inside the main useEffect

Something like this

```
useEffect(() => {
    const isPasskeyAvailable = await isPasskeyAvailable()

    if (isPassKeySupported === null) return;
    if (skipGetKeys) {
      setAuthenticated(false);
      setControllerState(false);
    } else if (controllerState !== false) {
.
.
.
```

That way we can remove the extra useEffect and useState

Reply:
We can't make a useEffect callback async directly so the only other option would be to wrap everything else in a promise or iife. Stylistically the current solution is slightly better than that imo

Reply:
I'm sorry, idd. 

I mean: 

```
useEffect(() => {
  const exampleName = async () => {
    const isPasskeyAvailable = await isPasskeyAvailable();

    if (isPasskeyAvailable === null) return;

    if (skipGetKeys) {
      setAuthenticated(false);
      setControllerState(false);
    } else if (controllerState !== false) {
      // ...
    }
  };

  exampleName();
}, [...deps]); 
```

Reply:
With the new changes on main it now makes sense to await `isPasskeyAvailable` so will make that change when I rebase this